pages/index.js

```diff
@@ -2,16 +2,56 @@ import useSWR from "swr";
 import Error from "@/components/Error";
 import Loading from "@/components/Loading";
 import EventList from "@/components/EventList";
+import FilterBar from "@/components/FilterBar";
+import { useState } from "react";
 
 export default function HomePage() {
-  const { data: events, isLoading, error } = useSWR(`/api/events`);
+  const [filter, setFilter] = useState({});
+
+  const {
+    data: events,
+    isLoading,
+    error,
+  } = useSWR(
+    `/api/events?queryCategory=${
+      filter?.category ? filter.category : ""
+    }&queryCity=${filter?.city ? filter.city : ""}`
+  );
+
+  function handleUpdateFilter(filterObject) {
+    setFilter((currentFilter) => {
+      return { ...currentFilter, ...filterObject };
+    });
+  }
+
+  const categories = events?.reduce(
+    (allCategories, curr) => [...allCategories, curr.category],
+    []
+  );
```

Comment:
Wouldn't this allow for duplicates if two events from the same category have the same?

A possible solution:

```js
function unique(array, key) {
  return [...new Set(array.map(x => x[key]))];
}

const events = [
  { id: 1, city: 'Berlin', category: 'a'},
  { id: 2, city: 'Berlin', category: 'a'},
  { id: 3, city: 'Hamburg', category: 'a'},
]

const cities = unique(events, 'city');
const categories = unique(events, 'category');
```

See the [Set docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set)